src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,269 @@
+#include "widget/wfindonwebmenu.h"
+
+#include <QDesktopServices>
+#include <QScreen>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/math.h"
+#include "util/qt.h"
+#include "util/widgethelper.h"
+
+namespace {
+
+QString composeActionText(const QString& prefix, const QString& trackProperty) {
+    return prefix + QStringLiteral(" | ") + trackProperty;
+}
+
+QString composeSearchQuery(const QString& trackAlbumOrTitle, const QString& artist) {
+    return trackAlbumOrTitle + QStringLiteral(" ") + artist;
+}
+
+QString composePrefixAction(WFindOnWebMenu::TrackSearchProperties trackSearchProperty) {
+    if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Title) {
+        return QObject::tr("Title");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Artist) {
+        return QObject::tr("Artist");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+        return QObject::tr("Title + Artist");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndAlbum) {
+        return QObject::tr("Album + Artist");
+    } else {
+        return QObject::tr("Album");
+    }
+}
+
+const QString searchUrlSoundCloudArtist = QStringLiteral("https://soundcloud.com/search/people?");
```

Comment:
Naming conventions are still violated -> `kSearchUrlSoundCloudArtist`

Reply:
Done :+1:. I have changed all constant URL to start with k.

What is k stands for? I am just asking to learn. I have checked the coding guidelines and google for naming conventions of constants couldn't find it.

Reply:
Makes sense. Thank you